File: client/src/types/chart.ts
export type ApiSeriesFormat = 'number' | 'percent';

export interface ApiChartColumn {
    name: string;
    format: ApiSeriesFormat;
}

export type ApiChartCell = number | null;

export interface ApiChartTable {
    columns: ApiChartColumn[];
    rows: ApiChartCell[][];
}

File: app/modules/mortality_ae/models/chart.py
from __future__ import annotations

from enum import StrEnum

from pydantic import BaseModel, ConfigDict
from typing import TypeAlias


class ApiSeriesFormat(StrEnum):
    NUMBER = "number"
    PERCENT = "percent"


class ApiChartColumn(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str
    format: ApiSeriesFormat = ApiSeriesFormat.NUMBER


ApiChartCell: TypeAlias = int | float | None


class ApiChartTable(BaseModel):
    model_config = ConfigDict(extra="forbid")

    columns: list[ApiChartColumn]
    rows: list[list[ApiChartCell]]
